cmd/gapit/validate_gpu_profiling.go

```diff
@@ -51,11 +51,15 @@ func (verb *validateGpuProfilingVerb) Run(ctx context.Context, flags flag.FlagSe
 	someDeviceFailed := false
 	for i, p := range devices {
 		fmt.Fprintf(stdout, "-- Device %v: %v --\n", i, p.ID.ID())
-		err = client.ValidateDevice(ctx, p)
+		res, err := client.ValidateDevice(ctx, p)
 		if err != nil {
 			fmt.Fprintf(stdout, "%v\n", log.Err(ctx, err, "Failed to validate device"))
 			someDeviceFailed = true
 			continue
+		} else if len(res.ValidationErrorMsg) > 0 {
```

Comment:
There are two kinds of "errors" here: 1) if we fail to run the validation for some reason (RPC, device communication issues, etc.) - e.g. we failed to run the validation - and 2) we successfully ran validation, but the device did not meet the checks and thus failed validation. Maybe instead of `ValidationErrorMsg` a better name for the field could be `ValidationFailureMsg`.

Now, "validation failure" could be made into an error type, yes, but I argued for it to be separate, because we have an established pattern, where RPCs return either a result or an error with the error indicating that something went wrong or the request was invalid. For validation, I see a "validation didn't pass" as a successful result to the "validate this device" request. Furthermore, reporting validation failures as errors would expand the error namespace too much.

I think the future plan is to also make the validation result more actionable in the UI. For example, we should probably give different advice if the device failed validation because it's too old (can't do anything about that) vs. when some counter failed to report the right data (try again, ensure the device is not otherwise busy, etc).

Reply:
To add: currently the validator returns a go `error` if validation doesn't pass. I think that will need to be changed in the future and it will return a `ValidationResult` - whatever that will look like - or an `error`.

Reply:
in that case, we should at least ensure that we do this consistently in our RPCs and it's somehow explicit about the need to check the result.

Maybe it already is. I haven't checked what the actual rpc is carrying. But if we have a clear error case in RPCs and this is consistent amongst the code, then I am ok with it.

Reply:
Yes, the other RPC result protos in the [service.proto](https://github.com/google/agi/blob/master/gapis/service/service.proto#L1336) include a similar structure
```
oneof res {
  Message result;
  Error err;
}
```